### Note
* Instructions have been included for each segment. You do not have to follow them exactly, but they are included to help you think through the steps.

In [1]:
# Import Dependencies
import pandas as pd

In [2]:
# File to Load (Remember to Change These)
file_to_load = "Resources/purchase_data.csv"

In [3]:
# Read Purchasing File and store into Pandas data frame
purchase_data = pd.read_csv(file_to_load)

## Player Count

* Display the total number of players


In [4]:
# Find unique players to calculate the totalnumber of players
unique_names = purchase_data["SN"].unique()
unique_names
# To find total number of players
total = len(unique_names)
total_players = pd.DataFrame({"Total Players": [total]})
total_players

Total Players
0            576

## Purchasing Analysis (Total)

* Run basic calculations to obtain number of unique items, average price, etc.


* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame


In [5]:
purchase_data.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [6]:
len(purchase_data)

780

In [7]:
#Run basic calculations to obtain number of unique items
number_of_unique_items = purchase_data["Item ID"].unique()
unique = len(number_of_unique_items)
unique

179

In [8]:
#Run basic calculations to obtain number of average price
avg_purchase_price = purchase_data["Price"].mean()
avg_purchase_price

3.050987179487176

In [9]:
# Run basic calculations to obtain number of purchases
total_number_of_purchases = purchase_data["Item Name"].count()
total_number_of_purchases

780

In [10]:
# Run basic calculations to obtain total revenue
total_revenue = purchase_data["Price"].sum()
total_revenue

2379.77

In [11]:
# Create a summary data frame to hold the results
purchasing_analysis = pd.DataFrame({"Number of Unique Items": [unique],
                                    "Average Price": [avg_purchase_price],
                                    "Number of Purchases": [total_number_of_purchases],
                                    "Total Revenue": [total_revenue]})
purchasing_analysis

Number of Unique Items  Average Price  Number of Purchases  Total Revenue
0                     179       3.050987                  780        2379.77

In [12]:
#Optional: give the displayed data cleaner formatting
purchasing_analysis = pd.DataFrame({"Number of Unique Items": [unique],
                                    "Average Price": '${:,.2f}'.format(avg_purchase_price),
                                    "Number of Purchases": total_number_of_purchases,
                                    "Total Revenue": '${:,.2f}'.format(total_revenue)})
purchasing_analysis

Number of Unique Items Average Price  Number of Purchases Total Revenue
0                     179         $3.05                  780     $2,379.77

## Gender Demographics

* Percentage and Count of Male Players


* Percentage and Count of Female Players


* Percentage and Count of Other / Non-Disclosed




In [13]:
# Create data frame with just SN and Gender information
df_gender = purchase_data[["SN","Gender"]]
df_gender


SN  Gender
0          Lisim78    Male
1      Lisovynya38    Male
2       Ithergue48    Male
3    Chamassasya86    Male
4        Iskosia90    Male
..             ...     ...
775     Aethedru70  Female
776         Iral74    Male
777     Yathecal72    Male
778        Sisur91    Male
779      Ennrian78    Male

[780 rows x 2 columns]

In [14]:
# Drop duplicated rows to keep only unique SN in new data frame
df_gender = df_gender.drop_duplicates(subset="SN", keep="first")
df_gender

SN  Gender
0          Lisim78    Male
1      Lisovynya38    Male
2       Ithergue48    Male
3    Chamassasya86    Male
4        Iskosia90    Male
..             ...     ...
773         Hala31    Male
774     Jiskjask80    Male
775     Aethedru70  Female
777     Yathecal72    Male
778        Sisur91    Male

[576 rows x 2 columns]

In [15]:
# Create a summary table with the count of unique SN by Gender using groupby function
summary_by_gender = df_gender.groupby(["Gender"]).count()
summary_by_gender

SN
Gender                    
Female                  81
Male                   484
Other / Non-Disclosed   11

In [16]:
# Rename the column SN as Total Count 
summary_by_gender=summary_by_gender.rename(columns={"SN":"Total Count"})
summary_by_gender

Total Count
Gender                            
Female                          81
Male                           484
Other / Non-Disclosed           11

In [17]:
# Percentage of players can be found by taking the count of players by gender and dividing with total number of players
percentage=summary_by_gender/summary_by_gender.sum() *100
percentage

Total Count
Gender                            
Female                   14.062500
Male                     84.027778
Other / Non-Disclosed     1.909722

In [18]:
# Add the percentage of players data as a newcolumn into the summary table
summary_by_gender["Percentage of Players"]= percentage
summary_by_gender

Total Count  Percentage of Players
Gender                                                   
Female                          81              14.062500
Male                           484              84.027778
Other / Non-Disclosed           11               1.909722

In [19]:
# Format the percentage data accordingly
summary_by_gender["Percentage of Players"]= summary_by_gender["Percentage of Players"].map('{:,.2f}%'.format)


In [20]:
# Sort the data frame by Total Count in descending order
summary_by_gender = summary_by_gender.sort_values("Total Count", ascending=False)
summary_by_gender

Total Count Percentage of Players
Gender                                                  
Male                           484                84.03%
Female                          81                14.06%
Other / Non-Disclosed           11                 1.91%

In [21]:
# To display the summary data frame
summary_by_gender[["Total Count", "Percentage of Players"]]

Total Count Percentage of Players
Gender                                                  
Male                           484                84.03%
Female                          81                14.06%
Other / Non-Disclosed           11                 1.91%


## Purchasing Analysis (Gender)

* Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. by gender




* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame

In [22]:
#Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. by gender

gender = purchase_data.groupby("Gender")
gender

In [23]:
purchase_count = gender["Age"].count()
purchase_count

Gender
Female                   113
Male                     652
Other / Non-Disclosed     15
Name: Age, dtype: int64

In [24]:
avg_purchase_price = gender["Price"].mean()
avg_purchase_price

Gender
Female                   3.203009
Male                     3.017853
Other / Non-Disclosed    3.346000
Name: Price, dtype: float64

In [25]:
purchase_total_per_person = gender["Price"].sum()
purchase_total_per_person 

Gender
Female                    361.94
Male                     1967.64
Other / Non-Disclosed      50.19
Name: Price, dtype: float64

In [26]:
avg_purchase_per_person = (gender["Price"].sum()/unique)
avg_purchase_per_person

Gender
Female                    2.022011
Male                     10.992402
Other / Non-Disclosed     0.280391
Name: Price, dtype: float64

In [27]:
purchase_data.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [28]:
# Create a summary data frame to hold the results
summary_df = pd.DataFrame({"Purchase count":purchase_count,"Average Purchase Price":avg_purchase_price,"Total Purchase Value":purchase_total_per_person ,"Average Purchase Total per Person":avg_purchase_per_person})

In [29]:
summary_df

Purchase count  Average Purchase Price  \
Gender                                                          
Female                            113                3.203009   
Male                              652                3.017853   
Other / Non-Disclosed              15                3.346000   

                       Total Purchase Value  Average Purchase Total per Person  
Gender                                                                          
Female                               361.94                           2.022011  
Male                                1967.64                          10.992402  
Other / Non-Disclosed                 50.19                           0.280391

In [30]:
# Optional: give the displayed data cleaner formatting
summary_df["Average Purchase Price"] = summary_df["Average Purchase Price"].map("${:.2f}".format)
summary_df["Total Purchase Value"]  = summary_df["Total Purchase Value"].map("${:,.2f}".format)
summary_df["Average Purchase Total per Person"] = summary_df["Average Purchase Total per Person"].map("${:.2f}".format)
summary_df.head()


Purchase count Average Purchase Price  \
Gender                                                         
Female                            113                  $3.20   
Male                              652                  $3.02   
Other / Non-Disclosed              15                  $3.35   

                      Total Purchase Value Average Purchase Total per Person  
Gender                                                                        
Female                             $361.94                             $2.02  
Male                             $1,967.64                            $10.99  
Other / Non-Disclosed               $50.19                             $0.28

In [31]:
# Display the summary data frame
summary_df.head()

Purchase count Average Purchase Price  \
Gender                                                         
Female                            113                  $3.20   
Male                              652                  $3.02   
Other / Non-Disclosed              15                  $3.35   

                      Total Purchase Value Average Purchase Total per Person  
Gender                                                                        
Female                             $361.94                             $2.02  
Male                             $1,967.64                            $10.99  
Other / Non-Disclosed               $50.19                             $0.28

## Age Demographics

* Establish bins for ages


* Categorize the existing players using the age bins. Hint: use pd.cut()


* Calculate the numbers and percentages by age group


* Create a summary data frame to hold the results


* Optional: round the percentage column to two decimal points


* Display Age Demographics Table


In [32]:
# Establish bins for ages
bins_ages = [0, 9.99, 14.99, 19.99, 24.99, 29.99, 34.99, 39.99, 44.99, 99.99]

bins_names = ["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40-44", "50+"]


In [33]:
# Categorize the existing players using the age bins. Hint: use pd.cut()
#purchase_data["Age Demographics"] = pd.cut(purchase_data["Age"], bins_ages , labels = bins_names)
# slice the data using pd.cut and  Categorize the existing players based on age_bins  
purchase_data["Age Group"] = pd.cut(purchase_data["Age"],bins_ages, labels=bins_names)
purchase_data.head()


Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price Age Group  
0  Extraction, Quickblade Of Trembling Hands   3.53     20-24  
1                          Frenzied Scimitar   1.56     40-44  
2                               Final Critic   4.88     20-24  
3                                Blindscythe   3.27     20-24  
4                                       Fury   1.44     20-24

In [34]:
# do a groupby on Age Group 
age_grouped = purchase_data.groupby("Age Group")

In [35]:
#calculate total count and percenatge of players by age category
unique_members = age_grouped["SN"].nunique()
Percent_of_players = (unique_members/576)*100

In [36]:
# create a DataFrame to hold the above results 
age_demographics = pd.DataFrame({"Percentage of players":Percent_of_players,"Total count":unique_members})

In [37]:
# Optional: round the percentage column to two decimal points
# change the index to none and do clean formatting 
age_demographics.index.name = None
age_demographics["Percentage of players"] = age_demographics["Percentage of players"].map("{:.2f}".format)
age_demographics

Percentage of players  Total count
<10                    2.95           17
10-14                  3.82           22
15-19                 18.58          107
20-24                 44.79          258
25-29                 13.37           77
30-34                  9.03           52
35-39                  5.38           31
40-44                  1.91           11
50+                    0.17            1

## Purchasing Analysis (Age)

* Bin the purchase_data data frame by age


* Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. in the table below


* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame

In [38]:
# Bin the purchase_data data frame by age
bins_ages = [0, 9.99, 14.99, 19.99, 24.99, 29.99, 34.99, 39.99, 44.99, 99.99]

bins_names = ["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40-44", "50+"]


In [39]:
# do calculations to obtain purchase count, avg. purchase price, total purchase value , average purchase per person
purchase_count = age_grouped["Age"].count()
purchase_count

<10       23
10-14     28
15-19    136
20-24    365
25-29    101
30-34     73
35-39     41
40-44     12
50+        1
Name: Age, dtype: int64

In [40]:
avg_purchase = age_grouped["Price"].mean()
avg_purchase

<10      3.353478
10-14    2.956429
15-19    3.035956
20-24    3.052219
25-29    2.900990
30-34    2.931507
35-39    3.601707
40-44    3.045000
50+      1.700000
Name: Price, dtype: float64

In [41]:
total_purchase_value = age_grouped["Price"].sum()
total_purchase_value

<10        77.13
10-14      82.78
15-19     412.89
20-24    1114.06
25-29     293.00
30-34     214.00
35-39     147.67
40-44      36.54
50+         1.70
Name: Price, dtype: float64

In [42]:
avg_purchase_person =(age_grouped["Price"].sum()/unique_members)
avg_purchase_person

<10      4.537059
10-14    3.762727
15-19    3.858785
20-24    4.318062
25-29    3.805195
30-34    4.115385
35-39    4.763548
40-44    3.321818
50+      1.700000
dtype: float64

In [43]:
# Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. in the table below
purchase_count = age_grouped["Age"].count()
avg_purchase = age_grouped["Price"].mean()
total_purchase_value = age_grouped["Price"].sum()
avg_purchase_person =(age_grouped["Price"].sum()/unique_members)

In [44]:
# create a DataFrame to hold the above results 
purchase_analysis_by_age = pd.DataFrame({"Purchase count":purchase_count,"Average Purchase Price":avg_purchase,
                           "Total Purchase value":total_purchase_value,"Average Purchase Total per person":avg_purchase_person})
purchase_analysis_by_age

Purchase count  Average Purchase Price  Total Purchase value  \
<10                23                3.353478                 77.13   
10-14              28                2.956429                 82.78   
15-19             136                3.035956                412.89   
20-24             365                3.052219               1114.06   
25-29             101                2.900990                293.00   
30-34              73                2.931507                214.00   
35-39              41                3.601707                147.67   
40-44              12                3.045000                 36.54   
50+                 1                1.700000                  1.70   

       Average Purchase Total per person  
<10                             4.537059  
10-14                           3.762727  
15-19                           3.858785  
20-24                           4.318062  
25-29                           3.805195  
30-34                           4.115385  
35-39                           4.763548  
40-44                           3.321818  
50+                             1.700000

In [45]:
# Optional: give the displayed data cleaner formatting
# do a clean formatting 
purchase_analysis_by_age["Average Purchase Price"] = purchase_analysis_by_age["Average Purchase Price"].map("${:.2f}".format)
purchase_analysis_by_age["Total Purchase value"]  = purchase_analysis_by_age["Total Purchase value"].map("${:,.2f}".format)
purchase_analysis_by_age["Average Purchase Total per person"] = purchase_analysis_by_age["Average Purchase Total per person"].map("${:.2f}".format)
purchase_analysis_by_age

Purchase count Average Purchase Price Total Purchase value  \
<10                23                  $3.35               $77.13   
10-14              28                  $2.96               $82.78   
15-19             136                  $3.04              $412.89   
20-24             365                  $3.05            $1,114.06   
25-29             101                  $2.90              $293.00   
30-34              73                  $2.93              $214.00   
35-39              41                  $3.60              $147.67   
40-44              12                  $3.04               $36.54   
50+                 1                  $1.70                $1.70   

      Average Purchase Total per person  
<10                               $4.54  
10-14                             $3.76  
15-19                             $3.86  
20-24                             $4.32  
25-29                             $3.81  
30-34                             $4.12  
35-39                             $4.76  
40-44                             $3.32  
50+                               $1.70

## Top Spenders

* Run basic calculations to obtain the results in the table below


* Create a summary data frame to hold the results


* Sort the total purchase value column in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the summary data frame



In [46]:
purchase_data.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price Age Group  
0  Extraction, Quickblade Of Trembling Hands   3.53     20-24  
1                          Frenzied Scimitar   1.56     40-44  
2                               Final Critic   4.88     20-24  
3                                Blindscythe   3.27     20-24  
4                                       Fury   1.44     20-24

In [47]:
# Top Spenders 
# Run basic calculations to obtain the results in the table below
# do a groupby on (SN-Screen Name) and assign it to a variable 
group_by_sn = purchase_data.groupby("SN")


In [48]:
# do calculations to obtain purchase count, avg. purchase price, total purchase value 
purchase_count =group_by_sn["Age"].count()
purchase_count

SN
Adairialis76     1
Adastirin33      1
Aeda94           1
Aela59           1
Aelaria33        1
                ..
Yathecal82       3
Yathedeu43       2
Yoishirrala98    1
Zhisrisu83       2
Zontibe81        3
Name: Age, Length: 576, dtype: int64

In [49]:
avg_purchase = group_by_sn["Price"].mean()
avg_purchase


SN
Adairialis76     2.280000
Adastirin33      4.480000
Aeda94           4.910000
Aela59           4.320000
Aelaria33        1.790000
                   ...   
Yathecal82       2.073333
Yathedeu43       3.010000
Yoishirrala98    4.580000
Zhisrisu83       3.945000
Zontibe81        2.676667
Name: Price, Length: 576, dtype: float64

In [50]:
total_purchase_value = group_by_sn["Price"].sum()
total_purchase_value

SN
Adairialis76     2.28
Adastirin33      4.48
Aeda94           4.91
Aela59           4.32
Aelaria33        1.79
                 ... 
Yathecal82       6.22
Yathedeu43       6.02
Yoishirrala98    4.58
Zhisrisu83       7.89
Zontibe81        8.03
Name: Price, Length: 576, dtype: float64

In [51]:
# create a DataFrame to hold the results 
topspenders_df = pd.DataFrame({"Purchase count":purchase_count,"Average Purchase Price":avg_purchase,
                           "Total Purchase value":total_purchase_value})


In [52]:
topspenders_df.head()

Purchase count  Average Purchase Price  Total Purchase value
SN                                                                        
Adairialis76               1                    2.28                  2.28
Adastirin33                1                    4.48                  4.48
Aeda94                     1                    4.91                  4.91
Aela59                     1                    4.32                  4.32
Aelaria33                  1                    1.79                  1.79

In [53]:
# Sort the total purchase value column in descending order 
topspenders_df = topspenders_df.sort_values("Total Purchase value", ascending = False)


In [54]:
topspenders_df.head()

Purchase count  Average Purchase Price  Total Purchase value
SN                                                                       
Lisosia93                 5                3.792000                 18.96
Idastidru52               4                3.862500                 15.45
Chamjask73                3                4.610000                 13.83
Iral74                    4                3.405000                 13.62
Iskadarya95               3                4.366667                 13.10

In [55]:
# Optional: give the displayed data cleaner formatting
# do a clean formatting 
topspenders_df["Average Purchase Price"] = topspenders_df["Average Purchase Price"].map("${:.2f}".format)
topspenders_df["Total Purchase value"]  = topspenders_df["Total Purchase value"].map("${:,.2f}".format)


In [56]:
# display the above results(Top Spenders sumamry table )
topspenders_df.head()

Purchase count Average Purchase Price Total Purchase value
SN                                                                     
Lisosia93                 5                  $3.79               $18.96
Idastidru52               4                  $3.86               $15.45
Chamjask73                3                  $4.61               $13.83
Iral74                    4                  $3.40               $13.62
Iskadarya95               3                  $4.37               $13.10

## Most Popular Items

* Retrieve the Item ID, Item Name, and Item Price columns


* Group by Item ID and Item Name. Perform calculations to obtain purchase count, average item price, and total purchase value


* Create a summary data frame to hold the results


* Sort the purchase count column in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the summary data frame



In [57]:
#Most Popular Items
# Retrieve the Item ID, Item Name, and Item Price columns (using .loc)  
popular_items_df = purchase_data.loc[:,["Item ID","Item Name","Price"]]
popular_items_df.head()

Item ID                                  Item Name  Price
0      108  Extraction, Quickblade Of Trembling Hands   3.53
1      143                          Frenzied Scimitar   1.56
2       92                               Final Critic   4.88
3      100                                Blindscythe   3.27
4      131                                       Fury   1.44

In [58]:
# Group by Item ID and Item Name. 
popular_items_df = purchase_data.groupby(["Item ID","Item Name"])


In [59]:
# Perform calculations to obtain purchase count, item price, and total purchase value
purchase_count = popular_items_df["Age"].count()
avg_purchase = popular_items_df["Price"].mean()
total_purchase_value = popular_items_df["Price"].sum()



In [60]:
# create a DataFrame to hold the results 
most_popular_items_df = pd.DataFrame({"Purchase count":purchase_count,"Average Purchase Price":avg_purchase,
                                   "Total Purchase value":total_purchase_value})


In [61]:
most_popular_items_df.head()

Purchase count  Average Purchase Price  \
Item ID Item Name                                                    
0       Splinter                         4                  1.2800   
1       Crucifer                         4                  2.9425   
2       Verdict                          6                  2.4800   
3       Phantomlight                     6                  2.4900   
4       Bloodlord's Fetish               5                  1.7000   

                            Total Purchase value  
Item ID Item Name                                 
0       Splinter                            5.12  
1       Crucifer                           11.77  
2       Verdict                            14.88  
3       Phantomlight                       14.94  
4       Bloodlord's Fetish                  8.50

In [62]:
# Sort the purchase count  column in descending order 
most_popular_items_df = most_popular_items_df.sort_values("Purchase count", ascending = False)


In [63]:
most_popular_items_df.head()

Purchase count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
132     Persuasion                                                 9   
108     Extraction, Quickblade Of Trembling Hands                  9   

                                                      Average Purchase Price  \
Item ID Item Name                                                              
92      Final Critic                                                4.614615   
178     Oathbreaker, Last Hope of the Breaking Storm                4.230000   
145     Fiery Glass Crusader                                        4.580000   
132     Persuasion                                                  3.221111   
108     Extraction, Quickblade Of Trembling Hands                   3.530000   

                                                      Total Purchase value  
Item ID Item Name                                                           
92      Final Critic                                                 59.99  
178     Oathbreaker, Last Hope of the Breaking Storm                 50.76  
145     Fiery Glass Crusader                                         41.22  
132     Persuasion                                                   28.99  
108     Extraction, Quickblade Of Trembling Hands                    31.77

In [64]:
# Optional: give the displayed data cleaner formatting
most_popular_items_df["Total Purchase value"]  = most_popular_items_df["Total Purchase value"].map("${:,.2f}".format)
most_popular_items_df["Average Purchase Price"] = most_popular_items_df["Average Purchase Price"].map("${:.2f}".format)


In [65]:
# display the Most Popular Items summary table 
most_popular_items_df.head()

Purchase count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
132     Persuasion                                                 9   
108     Extraction, Quickblade Of Trembling Hands                  9   

                                                     Average Purchase Price  \
Item ID Item Name                                                             
92      Final Critic                                                  $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm                  $4.23   
145     Fiery Glass Crusader                                          $4.58   
132     Persuasion                                                    $3.22   
108     Extraction, Quickblade Of Trembling Hands                     $3.53   

                                                     Total Purchase value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
132     Persuasion                                                 $28.99  
108     Extraction, Quickblade Of Trembling Hands                  $31.77

## Most Profitable Items

* Sort the above table by total purchase value in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the data frame



In [66]:
# Most Profitable Items
# Group by Item ID and Item Name. 
group_by_most_profitable = purchase_data.groupby(["Item ID","Item Name"])

In [67]:
# Perform calculations to obtain purchase count, item price, and total purchase value
purchase_count = group_by_most_profitable ["Age"].count()
avg_purchase = group_by_most_profitable ["Price"].mean()
total_purchase_value = group_by_most_profitable ["Price"].sum()


In [68]:
# create a DataFrame to hold the results 
most_popular_items_df = pd.DataFrame({"Purchase count":purchase_count,"Average Purchase Price":avg_purchase,
                                   "Total Purchase value":total_purchase_value})
most_popular_items_df.head()

Purchase count  Average Purchase Price  \
Item ID Item Name                                                    
0       Splinter                         4                  1.2800   
1       Crucifer                         4                  2.9425   
2       Verdict                          6                  2.4800   
3       Phantomlight                     6                  2.4900   
4       Bloodlord's Fetish               5                  1.7000   

                            Total Purchase value  
Item ID Item Name                                 
0       Splinter                            5.12  
1       Crucifer                           11.77  
2       Verdict                            14.88  
3       Phantomlight                       14.94  
4       Bloodlord's Fetish                  8.50

In [69]:
# Optional: give the displayed data cleaner formatting
#Sort the above table by total purchase value in descending order
most_popular_items_df = most_popular_items_df.sort_values("Total Purchase value", ascending = False)
most_popular_items_df["Total Purchase value"]  = most_popular_items_df["Total Purchase value"].map("${:,.2f}".format)
most_popular_items_df["Average Purchase Price"] = most_popular_items_df["Average Purchase Price"].map("${:.2f}".format)


In [70]:
# Display a preview of the data frame
# display the Most Profitable Items summary table 
most_popular_items_df.head()

Purchase count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
103     Singed Scalpel                                             8   

                                                     Average Purchase Price  \
Item ID Item Name                                                             
92      Final Critic                                                  $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm                  $4.23   
82      Nirvana                                                       $4.90   
145     Fiery Glass Crusader                                          $4.58   
103     Singed Scalpel                                                $4.35   

                                                     Total Purchase value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
82      Nirvana                                                    $44.10  
145     Fiery Glass Crusader                                       $41.22  
103     Singed Scalpel                                             $34.80

#  You must include a written description of three observable trends based on           the data.

1. Avg purchase price is $ 3.05 and major purchasers are people belonging to male gender.

2. Maximum percentage of players is 84.03% male and total purchase value is $1,967.64.

3. Maximum percentage of players belong to age group of 20-24 yrs amounting $ 1,114.06.

4. Top spenders are lisosia93 and idastidru52.

5. Most Popular Items and most Profitable Items are Final Critic and Oathbreaker,Last Hope of the Breaking Storm.
